### PR TITLE
fix: tls remove failed

### PR DESCRIPTION
**Pull request by gxthrj**
- [x] Fix: TLS remove failed .
- [x] Trigger retry when the nodes in upstream is empty .

### CODE REVIEW

**Review by membphis — COMMENTED** [review]
no test case?

File: pkg/ingress/endpoint/ep.go [hunk 1]
```diff
@@ -60,5 +60,12 @@ func BuildEps(ns, name string, port int) []*v1.Node {
 			}
 		}
 	}
+	if len(nodes) < 1 {
```
Comment (tokers) [inline pkg/ingress/endpoint/ep.go:63]: That's not a right behavior, if we really cannot collect the endpoint information, we shouldn't push this upstream to APISIX, instead, logging it and just throw it away.

Comment (gxthrj) [reply pkg/ingress/endpoint/ep.go:63]: ok, we do synchronization before sync, if there is no node, we will try again.

Comment (gxthrj) [reply pkg/ingress/endpoint/ep.go]: done

**Comment by gxthrj** [comment]
> no test case?

Added some test case for group field.

File: docs/FAQ.md [hunk 2]
```diff
@@ -38,3 +38,13 @@ This is because CRDs are generally declared in the file system, and Apply to ent
 So far apisix-ingress-controller doesn't support set admin_key for Apache APISIX, so when you deploy your APISIX cluster, admin_key should be removed from config.
 
 Note since APISIX have two configuration files, the first is config.yaml, which contains the user specified configs, the other is config-default.yaml, which has all default items, config items in these two files will be merged. So admin_key in both files should be removed. You can customize these two configuration files and mount them to APISIX deloyment.
+
+5. Failed to create route with `ApisixRoute`?
+
+When `apisix-ingress-controller` creates a route with CRD, it checks the `Endpoint` resources in Kubernetes (matched by namespace_name_port). If the corresponding endpoint information is not found, the route will not be created and wait for the next retry.
```
Comment (tokers) [inline docs/FAQ.md:44]: We should note that this is the limitation of APISIX.

Comment (gxthrj) [reply docs/FAQ.md:44]: Done

**Review by tokers — APPROVED** [review]

**Pull request merged**